lib/src/checkbox/Checkbox.tsx

```diff
@@ -189,106 +218,85 @@ const CheckboxContainer = styled.span`
-        outline-offset: -1px;
-      }
-    }
-    z-index: 1;
-    margin-left: ${(props) => (props.labelPosition === "before" && props.label ? props.theme.checkLabelSpacing : "0")};
-    margin-right: ${(props) => (props.labelPosition === "after" && props.label ? props.theme.checkLabelSpacing : "0")};
-    padding: 0px;
-    left: ${(props) => (props.labelPosition === "before" ? "unset" : "1px")};
-    right: ${(props) => (props.labelPosition === "before" ? "1px" : "unset")};
-  }
+const CheckboxContainer = styled.span`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 24px;
+  width: 24px;
 `;
 
-const CheckboxBlackBack = styled.span`
+type CheckboxProps = {
+  backgroundType: "dark" | "light";
+  isLabelHovered?: boolean;
+  checked: boolean;
+  disabled: boolean;
+};
+const Checkbox = styled.span<CheckboxProps>`
+  box-sizing: border-box;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 18px;
+  width: 18px;
+  border-radius: 2px;
+  border: solid 2px
+    ${(props) =>
+      props.disabled
+        ? getDisabledColor(props, "border")
+        : props.isLabelHovered
+        ? getNotDisabledColor(props, "hoverBorder")
+        : getNotDisabledColor(props, "border")};
   background-color: ${(props) =>
-    !props.checked
-      ? "transparent"
-      : props.disabled
-      ? getDisabledColor(props, "check")
-      : getNotDisabledColor(props, "check")};
-  width: 16px;
-  height: 16px;
-  position: absolute;
-  left: ${(props) => (props.labelPosition === "before" ? "unset" : "5px")};
-  right: ${(props) => (props.labelPosition === "before" ? "5px" : "unset")};
-  z-index: 0;
-  margin-left: ${(props) => (props.labelPosition === "after" ? "0px" : "")};
-  margin-right: ${(props) => (props.labelPosition === "before" ? "0px" : "")};
+    props.checked
+      ? props.disabled
+        ? getDisabledColor(props, "check")
+        : getNotDisabledColor(props, "check")
+      : "transparent"};
+  color: ${(props) =>
+    props.disabled
+      ? getDisabledColor(props, "background")
+      : props.isLabelHovered
+      ? getNotDisabledColor(props, "hoverBackground")
+      : getNotDisabledColor(props, "background")};
+
+  &:hover {
+    border: 2px solid
+      ${(props) => (props.disabled ? getDisabledColor(props, "border") : getNotDisabledColor(props, "hoverBorder"))};
+    background-color: ${(props) =>
+      props.checked
+        ? props.disabled
+          ? getDisabledColor(props, "check")
+          : getNotDisabledColor(props, "check")
+        : "transparent"};
+    color: ${(props) => (props.disabled ? getDisabledColor(props, "background") : getNotDisabledColor(props, "hoverBackground"))};
+  }
+  &:focus-visible {
+    outline: 2px solid
+      ${(props) => (props.backgroundType === "dark" ? props.theme.focusColorOnDark : props.theme.focusColor)};
+    outline-offset: 2px;
+  }
+  svg {
```

Comment:
Fix the svg size. Currently, the icon is overflowing due to its size of 18x18px, which, following the specs, should be 16x16px.

![MicrosoftTeams-image](https://user-images.githubusercontent.com/44321109/191232131-986f98d3-ee5f-4bc9-af4c-5ba739f03846.png)

![image](https://user-images.githubusercontent.com/44321109/191232229-27499ff0-700c-4b5b-aa6b-dfcc07bdeaa6.png)